interceptor/transportcc_interceptor.go

```diff
@@ -0,0 +1,41 @@
+package interceptor
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pion/webrtc/v3"
+)
+
+type TransportCCInterceptor struct {
```

Comment:
Don't know exactly how we will do it, but would be cool to have the `InterceptorFactory` pattern you mentioned.

Every Interceptor will have this boilerplate to Start/Delete and keep a list of PeerConnections. Just more code for users to grok (and where bugs can be)

Reply:
with this method the Interceptor is the factory, and the ReadWriter does what the interceptor did in my original propasal

i rewrote it to the interceptor2, it's much cleaner now